plugins/woocommerce/src/Internal/Logging/RemoteLogger.php

```diff
@@ -284,7 +288,16 @@ private function should_current_version_be_logged() {
 		}
 
 		// If the current version is the latest, we don't want to log errors.
-		return version_compare( WC()->version, $new_version, '>=' );
+		return version_compare( $this->get_wc_version(), $new_version, '>=' );
+	}
+
+	/**
+	 * Get the current WooCommerce version.
+	 *
+	 * @return string The current WooCommerce version.
+	 */
+	private function get_wc_version() {
+		return WC()->version;
```

Comment:
We should use `Constants::get_constant( 'WC_VERSION' );` instead.

I found that `WC()->version` is not safe because if the WC instance fails to initialize, it will cause a fatal error again.

Reply:
I was a bit conflicted about this one because it'll be reliant on Jetpack::Constants being available, which I'm not sure is more reliable than having WC() available. Furthermore it's a dependency for testing purposes, so I'm wondering if we should have a series of fallbacks for this instead ?

Maybe something like

```
function get_wc_version() {
    if (function_exists('WC') && method_exists(WC(), 'version')) {
        return WC()->version();
    }

    if (class_exists('\Automattic\Jetpack\Constants') && method_exists('\Automattic\Jetpack\Constants', 'get_constant')) {
        $wc_version = \Automattic\Jetpack\Constants::get_constant('WC_VERSION');
        if ($wc_version) {
            return $wc_version;
        }
    }

    if (defined('WC_VERSION')) {
        return WC_VERSION;
    }

    // Return null since none of the above worked
    return null;
}
```

Reply:
Good point! The [automattic/jetpack-constants](https://github.com/Automattic/jetpack-constants/tree/trunk) package is in our composer.json and has no dependencies, so I think it's safe to use. But fallbacks are a good idea! 👍 

I prefer using Jetpack::Constants first if available, then falling back to WC() if not. This can help prevent the fatal error loop when WC() fails to initialize. What do you think?

Reply:
I'll defer to your suggestion because I don't know better 😆